unittest/test_async.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "knowhere/index/VecIndexFactory.h"
+#include "knowhere/index/vector_index/ConfAdapterMgr.h"
+#include "knowhere/index/vector_index/adapter/VectorAdapter.h"
+#include "unittest/AsyncIndex.h"
+#include "unittest/Helper.h"
+#include "unittest/ThreadChecker.h"
+#include "unittest/utils.h"
+
+using ::testing::Combine;
+using ::testing::TestWithParam;
+using ::testing::Values;
+
+class AsyncIndexTest : public DataGen, public TestWithParam<knowhere::IndexType> {
+ protected:
+    void
+    SetUp() override {
+        index_type_ = GetParam();
+        if (index_type_ == knowhere::IndexEnum::INDEX_FAISS_BIN_IDMAP ||
+            index_type_ == knowhere::IndexEnum::INDEX_FAISS_BIN_IVFFLAT) {
+            Init_with_default(true);
+        } else {
+            Init_with_default();
+        }
+        conf_ = ParamGenerator::GetInstance().Gen(index_type_);
+        index_ = new knowhere::AsyncIndex(index_type_);
+    }
+
+    void
+    TearDown() override {
+    }
+
+ protected:
+    int32_t query_sum = 10;
```

Comment:
put it in an anonymous namespace outside this class and rephrase to `constexpr int32_t kQuerySum = 10;`